File: Freedom/微信/基本类库/Common/WXTableViewCell.m
//  WXTableViewCell.m
//  Freedom
// Created by Super
#import "WXTableViewCell.h"
@implementation WXTableViewCell
- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        _leftSeparatorSpace = 15.0f;
        _topLineStyle = TLCellLineStyleNone;
        _bottomLineStyle = TLCellLineStyleDefault;
    }
    return self;
}
- (void)setTopLineStyle:(TLCellLineStyle)topLineStyle{
    _topLineStyle = topLineStyle;
    [self setNeedsDisplay];
}
- (void)setBottomLineStyle:(TLCellLineStyle)bottomLineStyle{
    _bottomLineStyle = bottomLineStyle;
    [self setNeedsDisplay];
}
- (void)setLeftSeparatorSpace:(CGFloat)leftSeparatorSpace{
    _leftSeparatorSpace = leftSeparatorSpace;
    [self setNeedsDisplay];
}
- (void)setRightSeparatorSpace:(CGFloat)rightSeparatorSpace{
    _rightSeparatorSpace = rightSeparatorSpace;
    [self setNeedsDisplay];
}
- (void)drawRect:(CGRect)rect{
    [super drawRect:rect];
    CGContextRef context = UIGraphicsGetCurrentContext();
    CGContextSetLineWidth(context, 1 * 2);
    CGContextSetStrokeColorWithColor(context, [UIColor grayColor].CGColor);
    if (self.topLineStyle != TLCellLineStyleNone) {
        CGContextBeginPath(context);
        CGFloat startX = (self.topLineStyle == TLCellLineStyleFill ? 0 : _leftSeparatorSpace);
        CGFloat endX = self.frame.size.width - self.rightSeparatorSpace;
        CGFloat y = 0;
        CGContextMoveToPoint(context, startX, y);
        CGContextAddLineToPoint(context, endX, y);
        CGContextStrokePath(context);
    }
    if (self.bottomLineStyle != TLCellLineStyleNone) {
        CGContextBeginPath(context);
        CGFloat startX = (self.bottomLineStyle == TLCellLineStyleFill ? 0 : _leftSeparatorSpace);
        CGFloat endX = self.frame.size.width - self.rightSeparatorSpace;
        CGFloat y = self.frame.size.height;
        CGContextMoveToPoint(context, startX, y);
        CGContextAddLineToPoint(context, endX, y);
        CGContextStrokePath(context);
    }
}
@end

File: Freedom/微信/基本类库/Common/WXTableViewCell.swift
//
//  WXTableViewCell.swift
//  Freedom
//
//  Created by Chao Xue 薛超 on 2018/12/19.
//  Copyright © 2018 薛超. All rights reserved.
//

import Foundation
enum TLCellLineStyle : Int {
    case none
    case `default`
    case fill
}

class WXTableViewCell: UITableViewCell {
    var leftSeparatorSpace: CGFloat = 0.0 {
        didSet {
            setNeedsDisplay()
        }
    }
    var rightSeparatorSpace: CGFloat = 0.0 {
        didSet {
            setNeedsDisplay()
        }
    }
    var topLineStyle: TLCellLineStyle = .none {
        didSet {
             setNeedsDisplay()
        }
    }
    var bottomLineStyle: TLCellLineStyle = .none {
        didSet {
            setNeedsDisplay()
        }
    }
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        leftSeparatorSpace = 15.0
        topLineStyle = .none
        bottomLineStyle = .default
    }
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    override func draw(_ rect: CGRect) {
        super.draw(rect)
        guard let context = UIGraphicsGetCurrentContext() else { return }
        context.setLineWidth(1 * 2)
        context.setStrokeColor(UIColor.gray.cgColor)
        if topLineStyle != .none {
            context.beginPath()
            let startX: CGFloat = topLineStyle == .fill ? 0 : leftSeparatorSpace
            let endX: CGFloat = frame.size.width - rightSeparatorSpace
            let y: CGFloat = 0
            context.move(to: CGPoint(x: startX, y: y))
            context.addLine(to: CGPoint(x: endX, y: y))
            context.strokePath()
        }
        if bottomLineStyle != .none {
            context.beginPath()
            let startX: CGFloat = bottomLineStyle == .fill ? 0 : leftSeparatorSpace
            let endX: CGFloat = frame.size.width - rightSeparatorSpace
            let y = frame.size.height
            context.move(to: CGPoint(x: startX, y: y))
            context.addLine(to: CGPoint(x: endX, y: y))
            context.strokePath()
        }
    }
}
